### PR TITLE
OCPQE-22436 Add y version upgrade jobs for new e2e profiles

### PR DESCRIPTION
**Pull request by JianLi-RH**
Find new e2e profiles except below keywords:
`skipped_profiles = 'agent|alibaba|aro|destructive|disasterrecovery|hypershift|longduration|ocm|rhel|rosa|tp|ui|winc'`

```
[jianl@jianl-thinkpadt14gen4 release]$ e2e=ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml
[jianl@jianl-thinkpadt14gen4 release]$ upgrade=ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly-4.17-upgrade-from-stable-4.16.yaml
[jianl@jianl-thinkpadt14gen4 release]$ python ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py $e2e $upgrade
- as: aws-ipi-efa-pg-f5
- as: aws-ipi-localzone-byo-subnet-ovn-day2-f2
- as: aws-ipi-localzone-fips-f2
- as: aws-ipi-ovn-efs-fips-f14-cert-manager-custom-api-ingress
- as: aws-ipi-ovn-ipsec-f2-obo
- as: aws-ipi-ovn-ipsec-to-multiarch-f2		aws-ipi-ovn-ipsec-arm-mixarch-f28
- as: aws-ipi-private-shared-vpc-phz-sts-f2
- as: aws-ipi-proxy-fips-f14-fips-check		XXX
- as: aws-ipi-proxy-sdn-fips-f14
- as: aws-ipi-shared-vpc-phz-freeipa-f14
- as: aws-ipi-wavelength-byo-subnet-ovn-day2-f5
- as: aws-sc2s-ipi-disc-priv-fips-f2
- as: aws-usgov-ipi-disc-priv-f2
- as: azure-ipi-confidential-trustedlaunch-f5
- as: azure-ipi-to-multiarch-f2		azure-ipi-ovn-ipsec-arm-mixarch-f28
- as: azure-ipi-vmgenv1-f5
- as: azure-mag-ipi-fullyprivate-f2
- as: azure-stack-upi-f28
- as: baremetal-ipi-ovn-lvms-f14
- as: baremetal-upi-ovn-ipv4-f14	baremetal-upi-ovn-ipv4-arm-f28
- as: baremetalds-ipi-ovn-dualstack-primaryv6-f7
- as: baremetalds-ipi-ovn-ipv6-fips-f14
- as: gcp-ipi-mini-perm-custom-type-f28
- as: gcp-ipi-oidc-rt-fips-regen-cert-f14
- as: gcp-ipi-proxy-etcd-encryption-f28-security-profiles-operator
- as: gcp-ipi-secureboot-confidential-computing-f14
- as: gcp-ipi-to-multiarch-f14
- as: ibmcloud-ipi-mini-perm-f28
- as: ibmcloud-ipi-ovn-mtu-migrate-f28
- as: ibmcloud-ipi-private-fips-f28
- as: ibmcloud-ipi-proxy-private-rt-f14
- as: nutanix-ipi-boot-categories-project-rt-f28
- as: nutanix-ipi-compact-f28
- as: nutanix-ipi-image-registry-s3-f28
- as: nutanix-ipi-zones-external-lb-f28
- as: vsphere-ipi-amd-f7-netobserv	XXXXXX
- as: vsphere-ipi-multisubnets-elb-usermanaged-f28
- as: vsphere-ipi-ovn-dualstack-f28	vsphere-ipi-ovn-dualstack-privmaryv6-f28
- as: vsphere-ipi-ovn-dualstack-privmaryv6-f28
- as: vsphere-ipi-ovn-local-to-shared-gateway-mode-migration-f14
- as: vsphere-ipi-ovn-shared-to-local-gw-migration-f28
- as: vsphere-ipi-remote-worker-f28
- as: vsphere-ipi-sdn-migration-ovn-f14		XXXXX
- as: vsphere-ipi-static-ip-f28
- as: vsphere-ipi-template-usertags-f28
- as: vsphere-ipi-transit-masq-subnet-f28
- as: vsphere-ipi-zones-multisubnets-external-lb-f28
- as: vsphere-upi-disconnecting-network-f28
- as: vsphere-upi-encrypt-f28
- as: vsphere-upi-platform-external-f28
- as: vsphere-upi-platform-none-f28
- as: vsphere-upi-zones-f28
[jianl@jianl-thinkpadt14gen4 release]$ 



[jianl@jianl-thinkpadt14gen4 release]$ e2e=ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__arm64-nightly.yaml
[jianl@jianl-thinkpadt14gen4 release]$ upgrade=ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.16.yaml
[jianl@jianl-thinkpadt14gen4 release]$ python ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py $e2e $upgrade
- as: aws-ipi-ovn-ipsec-to-multiarch-f2
- as: azure-ipi-to-multiarch-f2
- as: baremetal-upi-ovn-ipv4-f14
- as: baremetalds-ipi-ovn-ipv4-f4
- as: gcp-ipi-to-multiarch-f14
[jianl@jianl-thinkpadt14gen4 release]$
```

### CODE REVIEW

**Comment by JianLi-RH** [comment]
Replace `- as: aws-ipi-ovn-ipsec-to-multiarch-f2` by `aws-ipi-ovn-ipsec-arm-mixarch-f28`
Replace `azure-ipi-to-multiarch-f2` by `azure-ipi-ovn-ipsec-arm-mixarch-f28`
Replace `baremetal-upi-ovn-ipv4-f14` by `baremetal-upi-ovn-ipv4-arm-f28`
Replace `vsphere-ipi-ovn-dualstack-f28` by `vsphere-ipi-ovn-dualstack-privmaryv6-f28`

Skip `aws-ipi-proxy-fips-f14-fips-check`
Skip `vsphere-ipi-amd-f7-netobserv`
Skip `vsphere-ipi-sdn-migration-ovn-f14`

Skip all agent jobs because https://redhat-internal.slack.com/archives/CF8SMALS1/p1712740822134299?thread_ts=1712650895.304669&cid=CF8SMALS1

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-nightly-4.17-upgrade-from-stable-4.16-aws-sc2s-ipi-disc-priv-fips-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-nightly-4.17-upgrade-from-stable-4.16-aws-usgov-ipi-disc-priv-f28

**Comment by JianLi-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-nightly-4.17-upgrade-from-stable-4.16-baremetalds-ipi-ovn-ipv4-f28

**Comment by JianLi-RH** [comment]
If there are no objections, please help review and merge this pr

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.16.yaml [hunk 1]
```diff
@@ -154,6 +155,25 @@ tests:
...
     workflow: cucushift-installer-rehearse-aws-ipi-disconnected-private-cco-manual-security-token-service
+- as: aws-sc2s-ipi-disc-priv-fips-f28
+  cron: 4 4 7 * *
+  steps:
+    cluster_profile: aws-sc2s-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      FIPS_ENABLED: "true"
+    test:
+    - chain: openshift-upgrade-qe-test
```
Comment (shellyyang1989) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.16.yaml:166]: should we use disconnected chain?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.16.yaml [hunk 2]
```diff
@@ -154,6 +155,25 @@ tests:
...
+    workflow: cucushift-installer-rehearse-aws-sc2s-ipi-disconnected-private
+- as: aws-usgov-ipi-disc-priv-f28
+  cron: 30 0 14 * *
+  steps:
+    cluster_profile: aws-usgov-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+    test:
+    - chain: openshift-upgrade-qe-test
```
Comment (shellyyang1989) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.16.yaml:175]: same as above

**Pull request merged**